docs/src/DevelopersGuide.md

```diff
@@ -0,0 +1,142 @@
+# Developer's Guide
+
+This doc is meant to guide any new developers with organization/formatting and common bugs associated with CloudMicrophysics.jl.
+
+## GitHub Pull Requests (PR)
+
+After creating a new branch, create a PR on GitHub.
+    There, you can comment the purpose of your branch and what needs to
+    be done to reach that goal. You can request reviewers to look over
+    and approve your work. 
+
+When creating PR's on GitHub, you will find that every
+    commit & push you make will go through checks. These checks are required
+    to pass before merging.
+The ``JuliaFormatter / format (pull_request)`` check
+    ensures consistent formatting. If this check fails, you can click on details
+    to see where and how your formatting needs to be fixed. If the correction
+    GitHub gives you looks the same as what you already have, it's likely that
+    you have an extra space at the end of the line.
+The ``documenter/deply`` check
+    can be useful when documentation is able to successfully build. Clicking on
+    details will bring up documentation of `CloudMicrophysics.jl` where everything
+    is processed neatly, including example boxes with plots. ``documenter/deploy``
+    and ``Documentation / docbuild (pull_request) `` focuses on documentation only.
+    Any errors will be related to documentation issues such as missing docstrings.
+The ``CodeCov`` and ``ci / ci 1.8.1 `` checks look at CloudMicrophysics library
+    as a whole. Oftentimes, if your files run fine locally, issues are related to
+    tests. See the `Tests` section below for debugging tips.
+
+Before merging, you can comment `bors try` on your PR to trigger a `build kite` check.
+    This will check if your code can run properly on GPU. If this check fails despite all
+    other checks succeeding, it is likely because a function used is not able to run on GPU.
+    Check your GPU tests for use of any out of place functions/commands (for example, `@warn`
+    will not work on GPU).
+
+Once someone has reviewed your PR and approved it,
+    make sure to squash all commits so that the PR only has 1 commit.
+    To merge to main, comment `bors r+` and wait for all checks to pass.
```

Comment:
```suggestion
Once someone has reviewed your PR and approved it,
    make sure to squash all commits so that the PR only has 1 commit.
We (still) use `bors` bot to manage PR queues.
To merge your PR into the `main` branch, comment `bors r+` and wait for all checks to pass.
```